src/context/instanceLoading.tsx

```diff
@@ -26,20 +20,14 @@ interface Props {
 }
 
 export const InstanceLoadingProvider: FC<Props> = ({ children }) => {
-  const [instanceStates, setInstanceStates] = useState(
-    new Map<string, LoadingTypes>(),
-  );
+  const instanceStates = new Map<string, LoadingTypes>();
```

Comment:
Does context trigger re-renders if we remove state here? i.e. does changes in the Map object trigger re-renders in children components subscribed to this context?

Reply:
Good point, this wasn't working properly when going to detail page / list page and back while a state change is in progress. Switched to a different approach just now.

Reply:
aah nice, by using the callback for `setState` calls you are guaranteed to use the correct previous state, nice! :+1: